smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructureGenerator.java

```diff
@@ -148,18 +161,27 @@ private void renderErrorStructure() {
 
         // Find symbol references with the "extends" property, and add SmithyException.
         writer.addImport("SmithyException", "__SmithyException", "@aws-sdk/smithy-client");
-        String extendsFrom = Stream.concat(
-                Stream.of("__SmithyException"),
+        String extendsFrom = Stream.concat(Stream.of("__SmithyException"),
                 symbol.getReferences().stream()
                         .filter(ref -> ref.getProperty(SymbolVisitor.IMPLEMENTS_INTERFACE_PROPERTY).isPresent())
-                        .map(SymbolReference::getAlias)
-                ).collect(Collectors.joining(", "));
+                        .map(SymbolReference::getAlias))
+                .collect(Collectors.joining(", "));
 
         writer.openBlock("export interface $L extends $L {", symbol.getName(), extendsFrom);
         writer.write("name: $S;", shape.getId().getName());
         writer.write("$$fault: $S;", errorTrait.getValue());
-        StructuredMemberWriter structuredMemberWriter = new StructuredMemberWriter(
-                model, symbolProvider, shape.getAllMembers().values());
+
+        Optional<RetryableTrait> retryableTrait = shape.getTrait(RetryableTrait.class);
```

Comment:
Shouldn't this just be:

```
HttpProtocolGeneratorUtils.writeRetryableTrait(writer, shape);
```

Reply:
Nope, because `StructureGenerator` requires `;` after the member is written while `Http<X>ProtocolGenerator` requires `,`

Reply:
Formatting removed in [commit](https://github.com/awslabs/smithy-typescript/pull/183/commits/c948214e41ce11721931182283ad491c8cb5adb0)

Reply:
Could that just be a `String` parameter to the method? I'd be worried about having separate implementations of how this is added becoming out of sync.

Reply:
Yup, the delimiter parameter can be just a String parameter to the method.
It has not been done in the past for other common params like `name`, `$$fault` though.

Will pass delimiter param for retryableTrait

Reply:
Added in [commit](https://github.com/awslabs/smithy-typescript/pull/183/commits/1349c612cc20d6776612537d5803d1402615b003)